src/zenhub.py

```diff
@@ -19,17 +19,17 @@ def main():
     repo_name = sys.argv[1]
     issue = sys.argv[2]
 
-    zen = ZenHub(repo_name=repo_name,
-                 issue=issue)
+    zen = ZenHub(repo_name=repo_name, issue=issue)
     print(json.dumps(zen.get_info()))
 
 
 class ZenHub():
 
-    def __init__(self, repo_name=None, issue=None):
+    def __init__(self, repo_name, issue):
         self.access_params = get_access_params(mgmnt_sys='zenhub')
         self.repo_name = repo_name
-        self.repo_id = self._get_repo_id(repo_name)
+        d = get_repo_id(repo_name)
+        self.repo_id = str(d['repo_id'])
```

Comment:
Something seems off with having get_repo_id return a dictionary that you have to access to get what you want. I understand wanting to have the response code available for return but why not have that be an optional arg, defaulting to False?

Reply:
I agree it could just return the number. Is the other info ever used?

Reply:
I think a function should always return a single data type, in this case it is a dictionary. I decided based on the happy path (status code 200) where the response has a dictionary. But also need to cover the failure when either `repo_name`, `org_name`, or none of them exists (not 200), in which case response doesn't have a `.json()`. So I decided to propagate any of them back to the caller (incl. `.reason`). The caller function (the ZenHub constructor) now raises `ValueError`.